sky/packages/sky/lib/rendering/object.dart

```diff
@@ -294,31 +360,45 @@ typedef void RenderObjectVisitor(RenderObject child);
 typedef void LayoutCallback(Constraints constraints);
 typedef double DimensionCallback(Constraints constraints);
 
+/// An object in the render tree
+///
+/// Render objects have a reference to their parent but do not commit to a model
+/// for their children.
 abstract class RenderObject extends AbstractNode implements HitTestTarget {
 
   // LAYOUT
 
-  // parentData is only for use by the RenderObject that actually lays this
-  // node out, and any other nodes who happen to know exactly what
-  // kind of node that is.
+  /// Data for use by the parent render object
+  ///
+  /// The parent data is used by the render object that lays out this object
+  /// (typically this object's parent in the render tree) to store information
+  /// relevant to itself and to any other nodes who happen to know exactly what
+  /// the data means. The parent data is opaque to the child.
   dynamic parentData; // TODO(ianh): change the type of this back to ParentData once the analyzer is cleverer
+
+  /// Override to setup parent data correctly for your children
   void setupParentData(RenderObject child) {
-    // override this to setup .parentData correctly for your class
     assert(debugCanPerformMutations);
     if (child.parentData is! ParentData)
       child.parentData = new ParentData();
   }
 
-  void adoptChild(RenderObject child) { // only for use by subclasses
-    // call this whenever you decide a node is a child
+  /// Called by subclases when they decide a render object is a child
+  ///
+  /// Only for use by subclasses
```

Comment:
Only for use by subclasses when changing their child lists; calling this in other cases will lead to an inconsistent tree and probably cause crashes.